File: rust/src/bin/283/283.move-zeroes.rs
/*
 * @lc app=leetcode id=283 lang=rust
 *
 * [283] Move Zeroes
 */


// @lc code=start
impl Solution {
    pub fn move_zeroes(nums: &mut Vec<i32>) {

        let mut swap_point = 0;

        for i in 0..nums.len() {
            if nums[i] != 0 {
                nums.swap(i, swap_point);
                swap_point += 1
            }
        }
    }
}
// @lc code=end

File: typescript/283/283.move-zeroes.ts
/* * @lc app=leetcode id=283 lang=typescript *
 * [283] Move Zeroes
 */

// @lc code=start
/**
 Do not return anything, modify nums in-place instead.
 */
function moveZeroes(nums: number[]): void {

    let write = 0;

    for (let i = 0; i < nums.length; i++) {

        if (nums[i] != 0) {
            const temp: number = nums[write];
            nums[write] = nums[i];
            nums[i] = temp;
            write++;
        }

    }

};
// @lc code=end
